ext/ffi/jit.rs

```diff
@@ -0,0 +1,247 @@
+  match ty {
+    NativeType::Void => "void",
+    NativeType::U8 => "unsigned char",
+    NativeType::U16 => "unsigned short",
+    NativeType::U32 => "unsigned int",
+    NativeType::U64 | NativeType::USize => "unsigned long",
+    NativeType::I8 => "char",
+    NativeType::I16 => "short",
+    NativeType::I32 => "int",
+    NativeType::I64 | NativeType::ISize => "long",
+    NativeType::F32 => "float",
+    NativeType::F64 => "double",
+    NativeType::Pointer | NativeType::Function => "void*",
+  }
+}
+
+macro_rules! impl_arg_int32 {
+  ($ctx: ident, $ty: ty) => {
+    paste::paste! {
+      unsafe extern "C" fn [<deno_ffi_ $ty>] (
+        info: *const v8::FunctionCallbackInfo,
+        index: i32,
+      ) -> $ty {
+        let scope = &mut v8::CallbackScope::new(&*info);
+        let info = v8::FunctionCallbackArguments::from_function_callback_info(info);
+        info.get(index).int32_value(scope).unwrap() as $ty
+      }
+
+      $ctx.add_symbol(
+        cstr!(stringify!([<deno_ffi_ $ty>])),
+        [<deno_ffi_ $ty>] as *const c_void,
+      );
+    }
+  };
+}
+
+macro_rules! impl_arg_uint32 {
+  ($ctx: ident, $ty: ty) => {
+    paste::paste! {
+      unsafe extern "C" fn [<deno_ffi_ $ty>] (
+        info: *const v8::FunctionCallbackInfo,
+        index: i32,
+      ) -> $ty {
+        let scope = &mut v8::CallbackScope::new(&*info);
+        let info = v8::FunctionCallbackArguments::from_function_callback_info(info);
+        info.get(index).uint32_value(scope).unwrap() as $ty
+      }
+
+      $ctx.add_symbol(
+        cstr!(stringify!([<deno_ffi_ $ty>])),
+        [<deno_ffi_ $ty>] as *const c_void,
+      );
+    }
+  };
+}
+
+macro_rules! impl_rv_int32 {
+  ($ctx: ident, $ty: ty) => {
+    paste::paste! {
+      unsafe extern "C" fn [<deno_rv_ $ty>] (
+        info: *const v8::FunctionCallbackInfo,
+        val: $ty,
+      ) {
+        let mut rv = v8::ReturnValue::from_function_callback_info(info);
+        rv.set_int32(val as i32);
+      }
+
+      $ctx.add_symbol(
+        cstr!(stringify!([<deno_rv_ $ty>])),
+        [<deno_rv_ $ty>] as *const c_void,
+      );
+    }
+  };
+}
+
+macro_rules! impl_rv_uint32 {
+  ($ctx: ident, $ty: ty) => {
+    paste::paste! {
+      unsafe extern "C" fn [<deno_rv_ $ty>] (
+        info: *const v8::FunctionCallbackInfo,
+        val: $ty,
+      ) {
+        let mut rv = v8::ReturnValue::from_function_callback_info(info);
+        rv.set_uint32(val as u32);
+      }
+
+      $ctx.add_symbol(
+        cstr!(stringify!([<deno_rv_ $ty>])),
+        [<deno_rv_ $ty>] as *const c_void,
+      );
+    }
+  };
+}
+
+pub struct Allocation {
+  pub addr: *mut c_void,
+  _ctx: Context,
+  _sym: Box<Symbol>,
+}
+
+pub(crate) struct Compiler {
+  ctx: Context,
+  c: String,
+}
+
+impl Compiler {
+  pub unsafe fn new() -> Result<Self, ()> {
+    let mut ctx = Context::new()?;
+    ctx.set_options(cstr!("-nostdlib"));
+
+    ctx.add_symbol(cstr!("deno_ffi_u8"), deno_ffi_u8 as *const c_void);
```

Comment:
This seems to be unnecessary, I think? The `impl_arg_uint32!(ctx, u8)` does this right?

Reply:
Yes, very much WIP.